src/components/DataSourceSelector.vue

```diff
@@ -0,0 +1,222 @@
+<template>
+  <section class="data-source-selector">
+    <div v-if="dataSources.length && !selectedDataSource">
+      <Select2
+        :dataSources="dataSources"
+        :selectedDataSource="selectedDataSource"
+        :customOptionView="formatDataSourceOption"
+        :customSearch="customDataSourceSearch"
+        :optionLabelKey="'name'"
+        :optionValueKey="'id'"
+        :selectWithGroups="!!otherDataSources.length"
+        @selectDataSource="setDataSource">
+        </Select2>
+      <span @click="() => { this.$emit('onDataSourceCreate') }" class="btn-link create-dataSource">Create new data source</span>
```

Comment:
Use kebab case for class names and IDs.

```suggestion
      <span @click="foo" class="btn-link create-data-source">Create new data source</span>
```

And use a function for event handlers.